lib/board.rb

```diff
@@ -69,13 +70,31 @@ def same_symbole?(array, mark)
   end
 
   def win_with_diagonal?(mark)
-    if board[0][0] == mark && board[1][1] == mark && board[2][2] == mark
-      true
-    elsif board[0][2] == mark && board[1][1] == mark && board[2][0] == mark
-      true
-    else
-      false
+    check_diag_one(mark) || check_diag_two(mark)
+  end
+
+  def check_diag_one(mark)
+    result = true
+
+    board.each_with_index do |value, index|
```

Comment:
This could be solved with `any?`:
`board.each_with_index.any? { | _, index| board[index][index] != mark }`
Or maybe
`[1..3].any? {|index| board[index][index] != mark }`